src/TorSharp/TorSharpProxy.cs

```diff
@@ -11,9 +11,19 @@
 
 namespace Knapcode.TorSharp
 {
+    public static class TorSharpProxyExtensions
+    {
+        public static async Task ConfigureAndStartAsync(this ITorSharpProxy proxy)
```

Comment:
I'm curious why you opted to make this an extension method instead of leaving the existing `ITorSharpProxy.ConfigureAndStartAsync()`. I think with this approach, consuming source is compatible but it would be a breaking change at runtime (i.e. a method not found exception on `TorSharpProxy.ConfigureAndStartAsync`).

I'd prefer to avoid runtime breaking changes even if it means the `ITorSharpProxy` bloats a little bit.

Reply:
Actually, I'm not 100% sure since the `TorSharpProxy.ConfigureAndStartAsync` concrete implementation still exists... but either way I think bloating the interface is fine unless I am missing a consideration you had.

Reply:
Well, the consideration was that if you were really implementing/mocking the interface, you'd need to actually provide an implementation of `ConfigureAndStartAsync` which is 100% the same in every potential implementation of the interface, since, it should just call ConfigureAsync then StartAsync. I like keeping interfaces to a minimum so you know what's absolutely the minimum required to implement, as opposed to implementing also usability overloads (or like this case, a composite behavior that doesn't actually care what the implementation of ITorSharpProxy really is).

Reply:
That rationale is reasonable. Thanks for talking me through it!